overlord/configstate/configstate.go

```diff
@@ -133,6 +134,22 @@ func Configure(st *state.State, snapName string, patch map[string]interface{}, f
 	return state.NewTaskSet(task)
 }
 
+// DefaultConfigure returns a taskset to apply the given default-configuration patch.
+func DefaultConfigure(st *state.State, snapName string) *state.TaskSet {
+	summary := fmt.Sprintf(i18n.G("Run default-configure hook of %q snap if present"), snapName)
+	hooksup := &hookstate.HookSetup{
```

Comment:
Is it intentional that `IgnoreError` and `TrackHookError` aren't set? I would assume that it's also valid to set them here but maybe we don't because it's optional?

Reply:
Yes it is intentional. The thinking, using `configure` as reference: `IgnoreError` and `TrackHookError` was only set for `doInstall` case for [snap "core"](https://github.com/snapcore/snapd/pull/12701/files/9b49a49eb0b197c4198f916e9e66742b2b3fd501#diff-9e832b3649d23b262a0cc4837c71c91a0e5d72ed1f2977cd7151b5a8e27da360R746). Core gets handled by [hijacked `configure` handler](https://github.com/snapcore/snapd/pull/12701/files/9b49a49eb0b197c4198f916e9e66742b2b3fd501#diff-c3ca63c3dae47699f1bf2cb3f45c6f128bad171832690efebb1a4fb544c18a30R56), but is by design [avoided for default-configure](https://github.com/snapcore/snapd/pull/12701/files/9b49a49eb0b197c4198f916e9e66742b2b3fd501#diff-9e832b3649d23b262a0cc4837c71c91a0e5d72ed1f2977cd7151b5a8e27da360R314) altogether. useDefaults is implicit to `default-configure`, so the default flags value of 0 in the Hook setup is therefore arguable correct.